test/lib/middleware.test.js

```diff
@@ -0,0 +1,69 @@
+const chai = require('chai')
+const proxyquire = require('proxyquire')
+const spies = require('chai-spies')
+
+chai.use(spies)
+const expect = chai.expect
+
+const CORRELATION_ID = 'correlation'
+const AUTH_TOKEN = 'authentication'
+
+const mockDomain = {

```

Comment:
This mock has global data, this is bad because one test might call the `run` function and on the next test `runCallback` would still have the previous callback. You should create the mock in a beforeEach block.